components/centraldashboard/public/components/manage-users-view_test.js

```diff
@@ -0,0 +1,213 @@
+/* eslint-disable max-len */
+import '@polymer/test-fixture/test-fixture';
+import 'jasmine-ajax';
+import {mockIronAjax, yeildForRequests} from '../ajax_test_helper';
+import {flush} from '@polymer/polymer/lib/utils/flush.js';
+
+import './dashboard-view';
+
+const FIXTURE_ID = 'manage-users-view-fixture';
+const MU_VIEW_SELECTOR_ID = 'test-manage-users-view';
+const TEMPLATE = `
+<test-fixture id="${FIXTURE_ID}">
+  <template>
+    <manage-users-view id="${MU_VIEW_SELECTOR_ID}"></manage-users-view>
+  </template>
+</test-fixture>
+`;
+const TEST_USER = 'test@kubeflow.org';
+const newNS = (namespace, role, user='test@kubeflow.org') => ({role, namespace, user});
+const oNs = newNS('ns1', 'owner');
+const generalNs = [oNs, newNS('ns2', 'contributor'), newNS('ns3', 'contributor')];
+
+describe('Manage Users View', () => {
+    let manageUsersView;
+
+    beforeAll(() => {
+        jasmine.Ajax.install();
+        const div = document.createElement('div');
+        div.innerHTML = TEMPLATE;
+        document.body.appendChild(div);
+    });
+
+    beforeEach(() => {
+        document.getElementById(FIXTURE_ID).create();
+        manageUsersView = document.getElementById(MU_VIEW_SELECTOR_ID);
+    });
+
+    afterEach(() => {
+        document.getElementById(FIXTURE_ID).restore();
+    });
+
+    afterAll(() => {
+        jasmine.Ajax.uninstall();
+    });
+
+    it('UI has loading and hidden sections when there\'s no data', () => {
+        expect(manageUsersView.shadowRoot.querySelector('.Acct-Info > .content').innerText)
+            .toBe('Loading...');
+        expect(manageUsersView.shadowRoot.querySelector('.Namespaces')
+            .hasAttribute('hidden')).toBe(true);
+        expect(manageUsersView.shadowRoot.querySelector('.Contributors')
+            .hasAttribute('hidden')).toBe(true);
+        expect(manageUsersView.shadowRoot.querySelector('.Cluster-Namespaces')
+            .hasAttribute('hidden')).toBe(true);
+    });
+
+    it('UI State should show user / contribs and unhide sections when namespace available', async () => {
+        const contribList = ['foo@kubeflow.org', 'bar@kubeflow.org'];
+        mockIronAjax(
+            manageUsersView.$.GetContribsAjax,
+            contribList,
+        );
+
+        manageUsersView.user = TEST_USER;
+        manageUsersView.ownedNamespace = oNs;
+        manageUsersView.namespaces = generalNs;
+
+        flush();
+        await yeildForRequests();
+
+        expect(manageUsersView.shadowRoot.querySelector('.Acct-Info > .content').innerText)
+            .toBe('test@kubeflow.org');
+        expect(manageUsersView.shadowRoot.querySelector('.Namespaces')
+            .hasAttribute('hidden')).toBe(false, 'Namespaces was still hidden');
```

Comment:
It looks like you're using the second argument as a failure message. I couldn't find any place in the [Docs](https://jasmine.github.io/api/3.4/matchers) where the second argument is described, but I do see that it will show the message as part of the output if the assertion fails.

Reply:
Yeah I found these really useful for test debuggability and resolving issues with new features.